win/pkgs/env/toolchain.ps1

```diff
@@ -169,8 +169,39 @@ if (-Not $Env:ROASTER_TOOLCHAIN_COMMITED)
 
     if (${Env:VSCMD_VER} -eq $null)
     {
-        ${VS_HOME} = & "${Env:ProgramFiles(x86)}/Microsoft Visual Studio/Installer/vswhere.exe" -latest -property installationPath
-        Invoke-Expression $($(cmd /c "`"${VS_HOME}/VC/Auxiliary/Build/vcvarsall.bat`" x64 10.0.16299.0 & set") -Match '^.+=' -Replace '^','${Env:' -Replace '=','}="' -Replace '$','"' | Out-String)
+        if (${Env:VS_HOME} -eq $null)
```

Comment:
User-provided path should only be considered after failure of auto-resolution, maybe as VS_EXTRA_HOME.
We don't want our downstream to have any hard-coded value when calling this script, because that means it's harder to ship suggested updates due to overriding.

Reply:
Yes. I found that vswhere is still installed with vs_buildtool. But your query string output nothing. Updating the query string should fix the issue.

Reply:
vswhere.exe -latest -products *  -requires Microsoft.VisualStudio.Component.VC.Tools.x86.x64 -property installationPath

see, https://github.com/boostorg/build/blob/a3bdb4fd98996d2564eed838b72046a1e99bf367/src/engine/vswhere_usability_wrapper.cmd